[BlazeChron] iP

BlazeChron opened the pull request:
### YAP YAP YAP

That's what you sound like

> "YAP YAP YAP YAP YAP YAP YAP YAP YAP YAP YAP YAP YAP YAP YAP YAP YAP YAP YAP YAP YAP" - You, yesterday. I'd know. I was there ([source](https://www.youtube.com/watch?v=r7l0Rq9E8MY))

Oh, it looks like you dropped this 📝

-------------
TO DO LIST
- [ ] YAP
- [x] YAP
- [ ] YAP
-------------

You should

1. Stop yapping to me
2. Download YapMeister
3. Yap to it instead

OMG it has a todolist feature too?
Say no to ~~physical~~ -> Go **digital!**

`return` from whence you came yapper!
- Get banished!
- Get banished!
- Get banished!

```python
#nooo not python in my java project!
def hello():
  print("hello world")
```

If you made it here, I'm sorry you read that. 🙇
Hope you had a laugh instead, have a nice day :)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

BlazeChron committed:
Renamed Duke to YapMeister and edited greeting

BlazeChron committed:
Added echoing to YapMeister

BlazeChron committed:
added Add and List functions

BlazeChron committed:
added Task class

BlazeChron committed:
added completed to Task and mark/unmark to YapMeister

BlazeChron committed:
changed Task to abstract and added three subclasses. Edited YapMeister to new Task inputs

BlazeChron committed:
added automated testing

BlazeChron committed:
Added error handling for invalid inputs for YapMeister, marking and Tasks

BlazeChron committed:
changed member variables to final

BlazeChron committed:
changed to use java collections ArrayList

BlazeChron committed:
added delete function to YapMeister

BlazeChron committed:
Edit current code to comply with Coding standard

Code does not comply with coding standard

I will lose marks.

Let's edit the code so it complies with the standard.

Also I did not read all the tasks so a commit to fit the standard is
supposed to be added later. I will make another commit then.

BlazeChron committed:
Update .gitignore ignore Task save file

BlazeChron committed:
Add create directory and task save file with error handling

BlazeChron committed:
Add Task saving and loading to YapMeister

TaskMeister starts with no tasks and cannot save tasks

It will be useful for users to be able to keep a record of tasks outside
of a session

Let's
Add file saving and loading for Tasks

BlazeChron committed:
Merge branch 'branch-Level-7'

BlazeChron committed:
Add LocalDate functionality to Deadline and Event

BlazeChron committed:
Merge branch 'branch-Level-8'

BlazeChron committed:
Add OOP classes

BlazeChron committed:
Separate YapMeister into more OOP parts
YapMeister had most logic within main file.

It is cleaner and more logical to separate this into separate classes.

Let's
* create Storage
* create TaskList
* create Parser
* create UI

BlazeChron committed:
Adjust test cases for automated testing

BlazeChron committed:
Merge branch 'branch-A-MoreOOP'

BlazeChron committed:
Remove unnecessary imports in YapMeister.java

BlazeChron committed:
Move classes into packages

BlazeChron committed:
Merge branch 'branch-A-Packages'

BlazeChron committed:
Merge branch 'add-gradle-support'

BlazeChron committed:
Add test classes for Deadline and Event

BlazeChron committed:
Merge branch 'branch-A-JUnit'

BlazeChron committed:
Add META-INF

BlazeChron committed:
Update javadocs for all files

BlazeChron committed:
Adjust to fulfil coding standard requirements

BlazeChron committed:
Add find feature to YapMeister

BlazeChron committed:
Merge branch 'branch-A-JavaDoc'

BlazeChron committed:
Merge branch 'branch-A-CodingStandard'

BlazeChron committed:
Merge branch 'branch-Level-9'

BlazeChron committed:
Fix Storage not using filepath
Storage is using a preset filepath instead of inputted filepath.

This does not allow user to change filepath through YapMeister object.

Let's fix Storage and allow for dynamic filepath creation.

BlazeChron committed:
Add javadocs to commit

BlazeChron committed:
Merge branch 'branch-A-Varargs'

BlazeChron committed:
Add Command rework to YapMeister

YapMeister relies on Parser to handle inputs.

This is very inflexible and Parser will become more complex.

Let's
* create new interface Command to represent user commands.
* separate the Parser commands into their own individual classes using
  Command interface.

BlazeChron committed:
Update javadocs for Command Rework

BlazeChron committed:
Merge branch 'branch-YapMeisterCommandRework'

BlazeChron committed:
Remove wildcard dependency from Parser

BlazeChron committed:
Add checkstyle and adjust styling

BlazeChron committed:
Merge branch 'branch-A-CheckStyle'

LinkesV commented on src/main/java/yapmeister/task/TaskList.java:
```diff
@@ -0,0 +1,76 @@
+     * @param t Task to be added.
+     */
+    public void addTask(Task t) {
+        this.tasks.add(t);
```
By the coding quality naming conventions, names that are 'too short' are not preferable, so you can change it to be task

LinkesV commented on src/main/java/yapmeister/task/TaskList.java:
```diff
@@ -0,0 +1,76 @@
+     * @param index Index of the task.
+     * @return Task at the index.
+     */
+    public Task get(int index) {
```
I think the this method name is a bit genric, it can be named getTask to better convey to the reader accurately what it does

LinkesV commented on src/main/java/yapmeister/task/ToDo.java:
```diff
@@ -0,0 +1,38 @@
+     * {@inheritDoc}
+     */
+    @Override
+    public String exportString() {
```
After looking through your code, almost all of your method names follow the code quality guidelines. Nicely done!

LinkesV commented on src/main/java/yapmeister/task/TaskList.java:
```diff
@@ -0,0 +1,76 @@
+        return this.tasks;
+    }
+
+    public ArrayList<Task> getFilteredArrayList(Predicate<? super Task> predicate) {
```
Interesting use of Predicate function from Java to implement the find command

LinkesV reviewed (commented):
Nicely done on following the code quality guidelines. Most of my comments were just nitpicking. Well done

BlazeChron committed:
Add GUI

BlazeChron committed:
Merge branch 'branch-Level-10'

BlazeChron committed:
Fix double message and saving bugs
YapMeister GUI does not save files after each input. Also does not
unmark properly.

This is bad as it does not fulfill the requirements.

Let's fix the code to return the functionality

BlazeChron committed:
Merge branch 'branch-Level-10'

BlazeChron committed:
Fix build.gradle to Launcher

build.gradle is not appropriately adjusted to the new JavaFX updates
that were pushed.

This means we cannot build a proper fat jar file.

Let's fix the build.gradle file.

BlazeChron committed:
Fix code quality

The code does not follow code quality guidelines.

It is harder to read.

Let's change it to adhere to the guidelines

BlazeChron committed:
Add assertions

There are no assertions for some assumptions.

Having assertions helps with debugging.

Let's add assertions to fix this.

BlazeChron committed:
Merge pull request #1 from BlazeChron/branch-A-CodeQuality

Fix code quality

BlazeChron committed:
Merge branch 'master' into branch-A-Assertions

BlazeChron committed:
Merge pull request #2 from BlazeChron/branch-A-Assertions

Add assertions

BlazeChron committed:
Fix merge commit

BlazeChron committed:
Merge pull request #3 from BlazeChron/branch-A-Assertions

Fix merge commit

BlazeChron committed:
Fix TaskList removed behaviour

TaskList old behaviour was removed.

getTask is a very useful method.

Let's re add the old method getTask.

BlazeChron committed:
Add exact find behaviour to Find Command

Find command only searches substrings.

Having an exact search option is useful.

Let's add a "-e" flag to use exact find.

BlazeChron committed:
Merge pull request #4 from BlazeChron/branch-C-BetterSearch

Add exact find behaviour to Find Command

BlazeChron committed:
Merge branch 'branch-C-BetterSearch'

BlazeChron committed:
Adjust UI

BlazeChron committed:
Merge branch 'branch-A-BetterGui'

BlazeChron committed:
Merge branch 'master' of https://github.com/BlazeChron/ip

BlazeChron committed:
Add credit for Command

BlazeChron committed:
Merge branch 'Giving-Credit'

BlazeChron committed:
Add UI image

BlazeChron committed:
Merge branch 'Add-UI-Image'

BlazeChron committed:
Remove UI.png

BlazeChron committed:
Add Ui.png

BlazeChron committed:
Fix ByeCommand not exiting program

BlazeChron committed:
Merge branch 'ByeCommand-adjust'

BlazeChron committed:
Add introduction to User Guide

BlazeChron committed:
Add table of contents User Guide

BlazeChron committed:
Add sections for each command in User Guide

BlazeChron committed:
Fix FindCommand match multiple terms not working
FindCommand only takes in first input before space.

This makes it hard for users to find multi word terms.

Let's add multiple terms to the FindCommand

BlazeChron committed:
Merge branch 'FindCommand-fix'

BlazeChron committed:
Update user guide with command summary

BlazeChron committed:
Add Quick Start guide

BlazeChron committed:
Fix typoes

BlazeChron committed:
Merge branch 'branch-A-UserGuide'
